### PR TITLE
Fixes #302 - procedure to rename property, rename label, rename reltype

### LINKED ISSUE #302: procedure to rename property, rename label, rename reltype

### PR DESCRIPTION
**Pull request by DanielBerton**
Indexes and constraints are managed, legacy indexes are not

### CODE REVIEW

File: src/main/java/apoc/refactor/rename/Rename.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+			});
+		}
+		if(rel != null){
+			Iterable<ConstraintDefinition> constraintsForRel = db.schema().getConstraints(RelationshipType.withName(rel));
+			constraintsForRel.forEach((c) -> {
+				constraints.add(c.toString());
+			});
+		}
+		if (prop != null) {
+			Iterable<ConstraintDefinition> constraintsForProps = db.schema().getConstraints();
```
Comment (jexp) [inline src/main/java/apoc/refactor/rename/Rename.java]: indexes too

File: src/main/java/apoc/refactor/rename/Rename.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+			constraintsForProps.forEach((c)-> {
+				c.getPropertyKeys().forEach((p) -> {
+					if (p.equals(prop)){
+						constraints.add(c.toString());
+					}
+				});
+			});
+		}
+		List<BatchAndTotalResult>targetLongList = iterate.collect(Collectors.toList());
```
Comment (jexp) [inline src/main/java/apoc/refactor/rename/Rename.java]: iterate.findFirst().orElse(null);

**Comment by jexp** [comment]
Looks good to me.

**Comment by jexp** [comment]
can you fix the two small things?

File: src/main/java/apoc/refactor/rename/Rename.java (hunk 1 above)
Comment (inserpio) [inline src/main/java/apoc/refactor/rename/Rename.java]: right

File: src/main/java/apoc/refactor/rename/Rename.java (hunk 2 above)
Comment (inserpio) [inline src/main/java/apoc/refactor/rename/Rename.java]: better

**Comment by inserpio** [comment]
@jexp ok to postpone manual and automatic indexes management?

**Comment by jexp** [comment]
@inserpio sure, we can even leave it to the user or make it an option

**Pull request merged**